test_suite/tests/derive.rs

```diff
@@ -907,3 +907,19 @@ fn replace_segments_works() {
 
     assert_type!(R, ty);
 }
+
+#[test]
+fn rename_field() {
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    pub struct S {
+        #[scale_info(rename = "type")]
+        r#type: bool,
```

Comment:
I'd like to see a more "dramatic" rename also in the test, eg "foo" renamed to "bar" (eg as a second field in this struct) and then a non-renamed field in the struct just to show that also ok in conjunciton with renamed ones